main.tf

```diff
@@ -14,107 +14,40 @@
 	VPC configuration
  *****************************************/
-resource "google_compute_network" "network" {
-  name                    = var.network_name
+module "vpc" {
+  source                  = "./modules/vpc"
+  network_name            = var.network_name
   auto_create_subnetworks = var.auto_create_subnetworks
   routing_mode            = var.routing_mode
-  project                 = var.project_id
+  project_id              = var.project_id
   description             = var.description
-}
-
-/******************************************
-	Shared VPC
- *****************************************/
-resource "google_compute_shared_vpc_host_project" "shared_vpc_host" {
-  count      = var.shared_vpc_host == "true" ? 1 : 0
-  project    = var.project_id
-  depends_on = [google_compute_network.network]
+  shared_vpc_host         = var.shared_vpc_host
 }
 
 /******************************************
 	Subnet configuration
  *****************************************/
-resource "google_compute_subnetwork" "subnetwork" {
-  for_each                 = local.subnets
-  name                     = each.value.subnet_name
-  ip_cidr_range            = each.value.subnet_ip
-  region                   = each.value.subnet_region
-  private_ip_google_access = lookup(each.value, "subnet_private_access", "false")
-  dynamic "log_config" {
-    for_each = lookup(each.value, "subnet_flow_logs", false) ? [{
-      aggregation_interval = lookup(each.value, "subnet_flow_logs_interval", null)
-      flow_sampling        = lookup(each.value, "subnet_flow_logs_sampling", null)
-      metadata             = lookup(each.value, "subnet_flow_logs_metadata", null)
-    }] : []
-    content {
-      aggregation_interval = log_config.value.aggregation_interval
-      flow_sampling        = log_config.value.flow_sampling
-      metadata             = log_config.value.metadata
-    }
-  }
-  network     = google_compute_network.network.name
-  project     = var.project_id
-  description = lookup(each.value, "description", null)
-  secondary_ip_range = [
-    for i in range(
-      length(
-        contains(
-        keys(var.secondary_ranges), each.value.subnet_name) == true
-        ? var.secondary_ranges[each.value.subnet_name]
-        : []
-    )) :
-    var.secondary_ranges[each.value.subnet_name][i]
-  ]
+module "subnets" {
+  source           = "./modules/subnets"
+  project_id       = var.project_id
+  network_name     = module.vpc.network_name
+  description      = var.description
+  subnets          = var.subnets
+  secondary_ranges = var.secondary_ranges
 }
 
 /******************************************
 	Routes
  *****************************************/
-resource "google_compute_route" "route" {
-  count                  = length(var.routes)
-  project                = var.project_id
-  network                = var.network_name
-  name                   = lookup(var.routes[count.index], "name", format("%s-%s-%d", lower(var.network_name), "route", count.index))
```

Comment:
We can preserve these auto-created route names I think, if we don't force the map interface on users.

Reply:
this original auto created name logic now exists again in the route submodule.